tests/models/siglip/test_tokenization_siglip.py

```diff
@@ -0,0 +1,552 @@
+        self.assertEqual(vocab_keys[1], "<s>")
+
+    def test_full_tokenizer(self):
+        tokenizer = SiglipTokenizer(SAMPLE_VOCAB)
+
+        tokens = tokenizer.tokenize("This is a test")
+        self.assertListEqual(tokens, ["▁this", "▁is", "▁a", "▁t", "est"])
+
+        self.assertListEqual(tokenizer.convert_tokens_to_ids(tokens), [66, 46, 10, 170, 382])
+
+        tokens = tokenizer.tokenize("I was born in 92000, and this is falsé.")
+        self.assertListEqual(
+            tokens,
+            [
+                SPIECE_UNDERLINE,
+                "i",
+                SPIECE_UNDERLINE + "was",
+                SPIECE_UNDERLINE + "b",
+                "or",
+                "n",
+                SPIECE_UNDERLINE + "in",
+                SPIECE_UNDERLINE + "",
+                "9",
+                "2",
+                "0",
+                "0",
+                "0",
+                SPIECE_UNDERLINE + "and",
+                SPIECE_UNDERLINE + "this",
+                SPIECE_UNDERLINE + "is",
+                SPIECE_UNDERLINE + "f",
+                "al",
+                "s",
+                "é",
+            ],
+        )
+        ids = tokenizer.convert_tokens_to_ids(tokens)
+        print("Ids:", ids)
+        self.assertListEqual(ids, [7, 23, 21, 84, 55, 24, 19, 7, 0, 602, 347, 347, 347, 12, 66, 46, 72, 80, 6, 0])
+
+        back_tokens = tokenizer.convert_ids_to_tokens(ids)
+        self.assertListEqual(
+            back_tokens,
+            [
+                SPIECE_UNDERLINE,
+                "i",
+                SPIECE_UNDERLINE + "was",
+                SPIECE_UNDERLINE + "b",
+                "or",
+                "n",
+                SPIECE_UNDERLINE + "in",
+                SPIECE_UNDERLINE + "",
+                "<unk>",
+                "2",
+                "0",
+                "0",
+                "0",
+                SPIECE_UNDERLINE + "and",
+                SPIECE_UNDERLINE + "this",
+                SPIECE_UNDERLINE + "is",
+                SPIECE_UNDERLINE + "f",
+                "al",
+                "s",
+                "<unk>",
+            ],
+        )
+
+    @cached_property
+    def siglip_tokenizer(self):
+        return SiglipTokenizer.from_pretrained("nielsr/siglip-base-patch16-224")
+
+    @cached_property
+    def siglip_tokenizer_fast(self):
+        return SiglipTokenizerFast.from_pretrained("nielsr/siglip-base-patch16-224")
+
+    def get_tokenizer(self, **kwargs) -> SiglipTokenizer:
+        return self.tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_rust_tokenizer(self, **kwargs) -> SiglipTokenizerFast:
+        return self.rust_tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def test_rust_and_python_full_tokenizers(self):
+        if not self.test_rust_tokenizer:
+            return
+
+        tokenizer = self.get_tokenizer()
+        rust_tokenizer = self.get_rust_tokenizer()
+
+        sequence = "I was born in 92000, and this is falsé."
```

Comment:
let's test the canonicalization as well with punctuation

Reply:
This one does contain punctuation right?

Reply:
Yep, but we should test § {}§. which has to be kept and fine some small edge cases like {,} will the , be removed then it's {} so not removed or all removed.  (of course the order is obvious for that simple example but if there is a special token in between then tokenize will happen first, then _tokenize will see { then }. Just adding a test to make sure we show what we expect

Reply:
I've updated the code as in OpenCLIP, which means that {} is no longer kept